hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorSupplier.java

```diff
@@ -104,7 +103,7 @@ private CompletableFuture<Traverser<Integer>> update(List<JetSqlRow> rows, IMap<
             assert row.getFieldCount() == 1;
             keys.add(row.get(0));
         }
-        return map.submitToKeys(keys, updaterSupplier.get(UntrustedExpressionEvalContext.from(evalContext)))
+        return map.submitToKeys(keys, updaterSupplier.get(evalContext))
```

Comment:
@k-jamroz what do you think?

Reply:
hmm, interesting. So it seems that the operation on partition owner succeeds, but fails on the replicas (IMap replicates updates via `Backup` operation).
It is strongly desired that execution on backups gives the same result as primary operation execution, otherwise we will have inconsistency between primary and replica. It might be resolved later by anti-entropy mechanisms, but that is last resort.
AFAIR, backup operations are executed on best-effort (due to mentioned anti-entropy), so exception in them may be ignored and it might not cause primary operation failure.

Reply:
> Try to find a way to set the context after deserialization of the UpdatingEntryProcessor

That would be the preferred solution, however we currently lack infrastructure for that.